### ISSUE TITLE
It's not using the correct interpreter

### ISSUE THREAD

**Issue by yueyinqiu** [microsoft/vscode-flake8#403]
## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): N/A, as the problem is that it's using a wrong interpreter
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): I tend to use uv, but it's using a Python interpreter elsewhere
- Operating system (and version): Both on Ubuntu (22.04.5 LTS) and Windows 10
- Version of tool extension you are using: 2025.2.0 (latest stable)

### Expected Behavior
I created a virtual environment with uv and selected it as the python interpreter in vscode. So I suppose the flake8 in this environment would be used by this plugin. (I have set `"flake8.importStrategy": "fromEnvironment"`, and verified that flake8 is truly installed.)

### Actual Behavior
However it's not using the one in the environment. I have further checked the logs and found this:

```
[info] Settings used to run Server:
[
    {
        "cwd": "/home/yueyinqiu/OurWork/LlmIll/local_mllm_server/llm_ill_local_mllm_server",
        "enabled": true,
        "workspace": "file:///home/yueyinqiu/OurWork/LlmIll/local_mllm_server/llm_ill_local_mllm_server",
        "args": [],
        "severity": {
            "E": "Error",
            "F": "Error",
            "I": "Information",
            "W": "Warning"
        },
        "path": [],
        "ignorePatterns": [],
        "interpreter": [
            "/home/yueyinqiu/.local/bin/python3.14"
        ],
        "importStrategy": "fromEnvironment",
        "showNotifications": "off"
    }
]

...

/home/yueyinqiu/.local/bin/python3.14 -m flake8 --version
```

Here `/home/yueyinqiu/.local/bin/python3.14` is not the one in the virtual environment. I think this is the problem. 

## Reproduction Steps:

```sh
mkdir test
cd test
uv init
uv sync
uv add flake8
```

and then open the directory, and create the `.vscode/settings.json` like:

```json
{
    // This setting is modified.
    "flake8.importStrategy": "fromEnvironment",

    // The following settings are kept default
    "flake8.args": [],
    "flake8.cwd": "${workspaceFolder}",
    "flake8.severity": {
        "E": "Error",
        "F": "Error",
        "I": "Information",
        "W": "Warning"
    },
    "flake8.path": [],
    "flake8.interpreter": [],
    "flake8.ignorePatterns": [],
    "flake8.enabled": true,

    // This setting is listed in README.md but seemly doesn't exist.
    "flake8.showNotification": "off"
}
```

then select the interpreter

<img width="481" height="95" alt="Image" src="https://github.com/user-attachments/assets/4aed288a-34b5-43cc-8504-b573dce966d4" />

then run flake8 restart server and will found that log:

<img width="635" height="110" alt="Image" src="https://github.com/user-attachments/assets/0c855540-f904-4548-9d5b-81feddc32bfd" />

## Logs:
<details>
  <summary>Click here for detailed logs</summary>

<code>
2026-02-26 07:23:21.246 [info] No interpreter found from setting flake8.interpreter
2026-02-26 07:23:21.246 [info] Getting interpreter from ms-python.python extension for workspace /home/yueyinqiu/OurWork/test
2026-02-26 07:23:21.290 [info] Interpreter from ms-python.python extension for /home/yueyinqiu/OurWork/test: /home/yueyinqiu/.local/bin/python3.14
2026-02-26 07:23:21.292 [info] Server: Stop requested
2026-02-26 07:23:21.292 [debug] Server State: Stopped
2026-02-26 07:23:21.292 [info] [Trace - 7:23:21 AM] Sending request 'shutdown - (2)'.
2026-02-26 07:23:21.297 [info] [Trace - 7:23:21 AM] Received response 'shutdown - (2)' in 5ms.
2026-02-26 07:23:21.297 [info] No result returned.

2026-02-26 07:23:21.297 [info] [Trace - 7:23:21 AM] Sending notification 'exit'.
2026-02-26 07:23:21.297 [info] No parameters provided.

2026-02-26 07:23:21.297 [info] No interpreter found from setting flake8.interpreter
2026-02-26 07:23:21.298 [info] Getting interpreter from ms-python.python extension for workspace /home/yueyinqiu/OurWork/test
2026-02-26 07:23:21.342 [info] Interpreter from ms-python.python extension for /home/yueyinqiu/OurWork/test: /home/yueyinqiu/.local/bin/python3.14
2026-02-26 07:23:21.348 [info] Server run command: /home/yueyinqiu/.local/bin/python3.14 /home/yueyinqiu/.vscode-server/extensions/ms-python.flake8-2025.2.0/bundled/tool/lsp_server.py
2026-02-26 07:23:21.348 [info] Server: Start requested.
2026-02-26 07:23:21.348 [debug] Server State: Starting
2026-02-26 07:23:21.820 [info] CWD Server: /home/yueyinqiu/OurWork/test
2026-02-26 07:23:21.820 [info] sys.path used to run Server:
   /home/yueyinqiu/.vscode-server/extensions/ms-python.flake8-2025.2.0/bundled/tool
   /home/yueyinqiu/.local/share/uv/python/cpython-3.14.2-linux-x86_64-gnu/lib/python314.zip
   /home/yueyinqiu/.local/share/uv/python/cpython-3.14.2-linux-x86_64-gnu/lib/python3.14
   /home/yueyinqiu/.local/share/uv/python/cpython-3.14.2-linux-x86_64-gnu/lib/python3.14/lib-dynload
   /home/yueyinqiu/.local/share/uv/python/cpython-3.14.2-linux-x86_64-gnu/lib/python3.14/site-packages
   /home/yueyinqiu/.vscode-server/extensions/ms-python.flake8-2025.2.0/bundled/libs
2026-02-26 07:23:21.820 [info] Settings used to run Server:
[
    {
        "cwd": "/home/yueyinqiu/OurWork/test",
        "enabled": true,
        "workspace": "file:///home/yueyinqiu/OurWork/test",
        "args": [],
        "severity": {
            "E": "Error",
            "F": "Error",
            "I": "Information",
            "W": "Warning"
        },
        "path": [],
        "ignorePatterns": [],
        "interpreter": [
            "/home/yueyinqiu/.local/bin/python3.14"
        ],
        "importStrategy": "fromEnvironment",
        "showNotifications": "off"
    }
]

2026-02-26 07:23:21.820 [info] Global settings:
{
    "cwd": "${workspaceFolder}",
    "enabled": false,
    "workspace": "/home/yueyinqiu",
    "args": [],
    "severity": {
        "E": "Error",
        "F": "Error",
        "I": "Information",
        "W": "Warning"
    },
    "path": [],
    "ignorePatterns": [],
    "interpreter": [],
    "importStrategy": "fromEnvironment",
    "showNotifications": "off"
}

2026-02-26 07:23:21.821 [info] /home/yueyinqiu/.local/bin/python3.14 -m flake8 --version
2026-02-26 07:23:21.822 [info] CWD Linter: /home/yueyinqiu/OurWork/test
2026-02-26 07:23:21.853 [info] 
7.1.1 (mccabe: 0.7.0, pycodestyle: 2.12.1, pyflakes: 3.2.0) CPython 3.14.2 on
Linux

2026-02-26 07:23:21.853 [info] Version info for linter running for /home/yueyinqiu/OurWork/test:
7.1.1 (mccabe: 0.7.0, pycodestyle: 2.12.1, pyflakes: 3.2.0) CPython 3.14.2 on
Linux

2026-02-26 07:23:21.853 [info] SUPPORTED flake8>=5.0.0
FOUND flake8==7.1.1

2026-02-26 07:23:21.857 [debug] Server State: Running
2026-02-26 07:23:21.861 [info] [Trace - 7:23:21 AM] Received notification 'window/logMessage'.
2026-02-26 07:23:21.861 [info] Params: {
    "type": 4,
    "message": "/home/yueyinqiu/.local/bin/python3.14 -m flake8 --format='%(row)d,%(col)d,%(code).1s,%(code)s:%(text)s' /home/yueyinqiu/OurWork/test/main.py"
}

2026-02-26 07:23:21.861 [info] /home/yueyinqiu/.local/bin/python3.14 -m flake8 --format='%(row)d,%(col)d,%(code).1s,%(code)s:%(text)s' /home/yueyinqiu/OurWork/test/main.py
2026-02-26 07:23:21.861 [info] [Trace - 7:23:21 AM] Received notification 'window/logMessage'.
2026-02-26 07:23:21.861 [info] Params: {
    "type": 4,
    "message": "CWD Linter: /home/yueyinqiu/OurWork/test"
}

2026-02-26 07:23:21.861 [info] CWD Linter: /home/yueyinqiu/OurWork/test
2026-02-26 07:23:21.868 [info] [Trace - 7:23:21 AM] Received notification 'textDocument/publishDiagnostics'.
2026-02-26 07:23:21.868 [info] Params: {
    "uri": "file:///home/yueyinqiu/OurWork/test/main.py",
    "diagnostics": []
}

2026-02-26 07:26:22.385 [info] [Trace - 7:26:22 AM] Sending request 'textDocument/codeAction - (1)'.
2026-02-26 07:26:22.385 [info] Params: {
    "textDocument": {
        "uri": "file:///home/yueyinqiu/OurWork/test/main.py"
    },
    "range": {
        "start": {
            "line": 0,
            "character": 0
        },
        "end": {
            "line": 0,
            "character": 0
        }
    },
    "context": {
        "diagnostics": [],
        "triggerKind": 2
    }
}

2026-02-26 07:26:22.403 [info] [Trace - 7:26:22 AM] Received response 'textDocument/codeAction - (1)' in 18ms.
2026-02-26 07:26:22.403 [info] Result: []
</code>
</details>

## Outcome When Attempting Debugging Steps:
Did running it from the command line work? N/A, the command itself is executed with a wrong interpreter.

## Extra Details
1. Manually setting the `flake8.interpreter` to `.venv/bin/python3.14` would work. (But I expect that the default behavior should not be like this.)
2. It worked fine before, so I'm suspecting whether it's caused by a recent Python extension update. (However, downgrading did not solve the problem to me.)

This plugin has been a great help to me. Thanks in advance.
